framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2429,68 +2429,79 @@ private static final boolean validType(TypeMirror type) {
         }
     }
 
-    /** Sets indexTypes and indexDeclAnnos by side effect, just before returning. */
-    protected void buildIndexTypes() {
-        if (this.indexTypes != null || this.indexDeclAnnos != null) {
-            ErrorReporter.errorAbort("AnnotatedTypeFactory.buildIndexTypes called more than once");
+    /**
+     * Parses the stub files in the following order: <br>
+     * 1. jdk.astub in the same directory as the checker, if it exist and ignorejdkastub option is not supplied <br>
+     * 2. flow.astub in the same directory as BaseTypeChecker <br>
+     * 3. Stub files provide via stubs compiler option <br>
+     * 4. Stub files provide via stubs system property <br>
+     * 5. Stub files provide via stubs environment variable <br>
+     * 6. Stub files listed in @Stubfiles annotation on the checker; must be in same directory as the checker
```

Comment:
Can we also document how ordering matters, i.e. later entries for the same element overwrite previous ones?
If so, should 6. `@Stubfiles on checker` be moved earlier, to allow adaptations by user-provided options?